packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -594,6 +594,10 @@ typedef void StateSetter(VoidCallback fn);
 ///    [BuildContext] or the widget, which are available as the [context] and
 ///    [config] properties, respectively, when the [initState] method is
 ///    called.
+///  * The framework calls [dependenciesChanged]. Subclasses of [State] should
+///    override [dependenciesChanged] to perform initialization involving
+///    [InheritedWidget]s. This method will be called again if the inherited
+///    widgets change.
```

Comment:
It might be worth pointing out that inherited widgets implicitly (or we consider them to have implicitly changed) when a widget moves in the tree.